scripts/gen_images.py

```diff
@@ -0,0 +1,455 @@
+        dir_path = og_images_dir / dir_name
+        if not dir_path.exists():
+            continue
+
+        html_content += f"<div class='image-container'>\n<h2>{section_title}</h2>\n<div class='image-grid'>\n"
+
+        # Get sample images (limit to 50 per category to keep page size reasonable)
+        sample_images = []
+        for root, _, files in os.walk(dir_path):
+            image_files = [
+                f for f in files if f.lower().endswith((".png", ".jpg", ".jpeg"))
+            ]
+            sample_paths = [os.path.join(root, f) for f in image_files[:50]]
+            sample_images.extend(sample_paths)
+
+        # Add image cards to HTML
+        for img_path in sample_images:
+            rel_path = os.path.relpath(img_path, start=og_images_dir.parent.parent)
+            html_content += f"""    <div class='image-card'>
+        <img src='/{rel_path}' alt='OG Image'>
+        <div class='image-caption'>{os.path.basename(img_path)}</div>
+    </div>\n"""
+
+        html_content += "</div>\n</div>\n"
+
+    html_content += """</body>
+</html>"""
+
+    # Write the HTML file
+    with open(og_images_dir / "index.html", "w", encoding="utf-8") as f:
+        f.write(html_content)
+
+
+def main():
+    start_time = time.time()
+    conn = get_db_connection()
+
+    # Create output directories
+    og_images_dir = Path("public/images/og-images")
+    create_directory_if_not_exists(og_images_dir)
+    create_directory_if_not_exists(og_images_dir / "narrators")
+    create_directory_if_not_exists(og_images_dir / "hadiths")
+    create_directory_if_not_exists(og_images_dir / "chapters")
+
+    # Generate default image for fallback
+    print("Generating default OG image...")
+    generate_default_og_image(og_images_dir / "og-default.png")
+
+    # Generate narrator OG images - ONLY for narrators with hadiths
+    print("Generating narrator OG images (only for narrators in hadith chains)...")
+
+    # Get only narrators who appear in chains
+    narrators = conn.execute(
+        """
+        SELECT DISTINCT r.*
+        FROM rawis r
+        JOIN hadith_chains hc ON r.scholar_indx = hc.scholar_indx
+        ORDER BY r.scholar_indx
+    """
+    ).fetchall()
+
+    narrator_count = len(narrators)
+    print(f"Found {narrator_count} narrators with hadiths (out of total rawis)")
+
+    narrator_columns = [
+        desc[0] for desc in conn.execute("SELECT * FROM rawis LIMIT 1").description
+    ]
+
+    for narrator in tqdm(narrators):
+        narrator_dict = dict(zip(narrator_columns, narrator))
+        # Sanitize name for use as filename
+        sanitized_name = str(narrator_dict["name"]).replace("/", "-").replace("\\", "-")
+        output_path = og_images_dir / "narrators" / f"{sanitized_name}.png"
+        try:
+            generate_narrator_og_image(narrator_dict, output_path)
+        except Exception as e:
+            print(f"Error generating OG image for narrator {sanitized_name}: {e}")
```

Comment:
The generation is embarrassingly parallel, could you parallelize it so we maintain the build time below five mintues.